design/FeaturesWithLimits.md

```diff
@@ -0,0 +1,124 @@
+  - *Available (7)*: OK, enabled with limit set to default (5)
+  - *Available (5)*: OK, enabled with limit set to default (5)
+  - *Unavailable*:   error, feature not available
+  - *Unimplemented*: error, feature not available
+- `adapter.requestDevice({ requiredFeatures: ["new-feature"],  requiredLimits: { newLimit: adapter.limits.newLimit } })`
+  - *Available (7)*: OK, enabled with limit set to 7
+  - *Available (5)*: OK, enabled with limit set to 5
+  - *Unavailable*:   error, feature not available
+  - *Unimplemented*: error, feature not available
+- `adapter.requestDevice({ requiredFeatures: ["new-feature"],  requiredLimits: { newLimit:                       5 } })`
+  - *Available (7)*: OK, enabled with limit set to 5
+  - *Available (5)*: OK, enabled with limit set to 5
+  - *Unavailable*:   error, feature not available, limit key not available
+  - *Unimplemented*: error, feature not available, limit key not available
+- `adapter.requestDevice({ requiredFeatures: ["new-feature"],  requiredLimits: { newLimit:                       7 } })`
+  - *Available (7)*: OK, enabled with limit set to 7
+  - *Available (5)*: error, limit value not available
+  - *Unavailable*:   error, feature not available, limit key not available
+  - *Unimplemented*: error, feature not available, limit key not available
+- `adapter.requestDevice({ requiredFeatures: adapter.features, requiredLimits: { newLimit:               undefined } })`
+  - *Available (7)*: OK, enabled with limit set to default (5)
+  - *Available (5)*: OK, enabled with limit set to default (5)
+  - *Unavailable*:   OK, neither limit nor feature is requested
+  - *Unimplemented*: OK, neither limit nor feature is requested
+- `adapter.requestDevice({ requiredFeatures: adapter.features, requiredLimits: { newLimit: adapter.limits.newLimit } })`
+  - *Available (7)*: OK, enabled with limit set to 7
+  - *Available (5)*: OK, enabled with limit set to 5
+  - *Unavailable*:   OK, neither limit nor feature is requested
+  - *Unimplemented*: OK, neither limit nor feature is requested
+- `adapter.requestDevice({ requiredFeatures: adapter.features, requiredLimits: { newLimit:                       5 } })`
+  - *Available (7)*: OK, enabled with limit set to 5
+  - *Available (5)*: OK, enabled with limit set to 5
+  - *Unavailable*:   error, limit key not available
+  - *Unimplemented*: error, limit key not available
+- `adapter.requestDevice({ requiredFeatures: adapter.features, requiredLimits: { newLimit:                       7 } })`
+  - *Available (7)*: OK, enabled with limit set to 7
+  - *Available (5)*: error, limit value not available
+  - *Unavailable*:   error, limit key not available
+  - *Unimplemented*: error, limit key not available
+
+TODO: Open questions:
+- Should the device be "reified" such that when the feature is not enabled, the limit will be shown as `undefined`?
+    - We already have to reify such that when the feature is enabled, the limit will be shown as `5`.
```

Comment:
Given that above we state that an `undefined` limit is a signal that the adapter doesn't support the feature I think it makes sense to reify to `undefined` on the device side as well when the feature isn't requested.

That DOES lead to a somewhat strange interaction where I can request a specific limit but not the feature, at which point my requested limit gets changed to `undefined`. I don't think we have any other instances right now where we change an explicitly provided limit, but it doesn't seem too objectionable in this case. (And it probably goes without saying that requesting a limit without the associated feature is worth a console warning.)